### PR TITLE
Align Spake2 protocol message type definition with spec

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
Currently, transport layer still use the internal temporary message type definition for protocol Spake2

 #### Summary of Changes
Align Spake2 protocol message type definition with spec 0.7.

### CODE REVIEW

File: src/transport/SecurePairingSession.cpp [hunk 1]
```diff
@@ -181,23 +181,21 @@ CHIP_ERROR SecurePairingSession::WaitForPairing(uint32_t mySetUpPINCode, uint32_
...
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
     System::PacketBufferHandle msgBuf;
 
     PayloadHeader payloadHeader;
 
-    payloadHeader
-        .SetMessageType(msgType) //
-        .SetProtocolID(Protocols::kProtocol_SecureChannel);
+    payloadHeader.SetMessageType(static_cast<uint8_t>(msgType)).SetProtocolID(Protocols::kProtocol_SecureChannel);
```
Comment (andy31415) [inline src/transport/SecurePairingSession.cpp:198]: nit: it was slightly easier to read before even though // was awkward

**Review by andy31415 — APPROVED** [review]

**Review by pan-apple — COMMENTED** [review]
LGTM. I have this PR which updates the pairing state machine, and associated message types: https://github.com/project-chip/connectedhomeip/pull/4166.

Maybe you can rebase the current PR on top of #4166 once it merges. Or, if this one merges first, I can rebase the other PR.

**Comment by yufengwangca** [comment]
> LGTM. I have this PR which updates the pairing state machine, and associated message types: #4166.
> 
> Maybe you can rebase the current PR on top of #4166 once it merges. Or, if this one merges first, I can rebase the other PR.

Got it, thanks!

**Review by jelderton — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@yufengwangca Please merge on top of #4166  ?

**Review by mspang — APPROVED** [review]

**Comment by yufengwangca** [comment]
> @yufengwangca Please merge on top of #4166 ?

Done

File: src/transport/SecurePairingSession.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ void SecurePairingSession::Clear()
     memset(&mPoint[0], 0, sizeof(mPoint));
     memset(&mWS[0][0], 0, sizeof(mWS));
     memset(&mKe[0], 0, sizeof(mKe));
-    mNextExpectedMsg = Spake2pMsgType::kSpake2pMsgError;
+    mNextExpectedMsg = Protocols::SecureChannel::MsgType::PASE_Spake2pError;
```
Comment (bzbarsky-apple) [inline src/transport/SecurePairingSession.cpp:66]: I wonder whether `using namespace Protocols::SecureChannel;` might be helpful to reduce the typing.  Or `using MsgType = Protocols::SecureChannel::MsgType`?

**Review by bzbarsky-apple — APPROVED** [review]

File: src/transport/SecurePairingSession.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/transport/SecurePairingSession.cpp:66]: Followup for this one as desired.

Comment (yufengwangca) [reply src/transport/SecurePairingSession.cpp:66]: > using MsgType = Protocols::SecureChannel::MsgType

Now we have taken out the prefix of MsgType, Then how we differentiate "using MsgType = Protocols::Common::MsgType" and "using MsgType = Protocols::SecureChannel::MsgType" ?

Comment (bzbarsky-apple) [reply src/transport/SecurePairingSession.cpp:66]: If we have to deal with both of them here, then we can't do that, of course.  In that case maybe the best thing to do is `using namespace chip::Protocols;` or `namespace SecureChannel = Protocols::SecureChannel;`.  Again, just to make things a bit shorter; it's not a big deal either way.

**Pull request merged**